### ISSUE TITLE
Add code coverage in CI

### ISSUE THREAD

**Issue by spacebear21** [payjoin/rust-payjoin#381]
Ideally we should pick a tool that shows visual line-by-line code coverage, so we can easily determine which code paths need more robust testing.

**Comment by DanGould** [comment]
I admit I have no experience with these tools. Any suggestions?

**Comment by spacebear21** [comment]
I used [coverage.py](https://coverage.readthedocs.io/en/coverage-5.0.4/index.html) with GitLab in the past which was nice but I'm not really familiar what the rust/GitHub options. I talked to @0xBEEFCAF3 about it briefly last week and he mentioned [codecov](https://github.com/marketplace/codecov) (I'm pretty sure it was codecov...). rust-bitcoin uses coveralls, e.g. https://github.com/rust-bitcoin/rust-bitcoin/pull/3481#issuecomment-2416306047 

Not a high priority addition but I noticed it in the README milestones and it might help highlight untested code paths in PR reviews, so I wanted to track it somewhere.
